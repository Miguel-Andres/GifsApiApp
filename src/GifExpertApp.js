import React,{useState} from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";


export default function GifExpertApp({defaultCategories = []}) {
    const [categories, setCategories] = useState(defaultCategories)
  // const categories = ;


 

  return (
    <>
      <h2>GifExpertApp</h2>

      <AddCategory setCategories={setCategories} />

      <hr />
    
    

      <ol>
        {categories.map((item,i) => (
          <GifGrid key={item} category={item} />
        ))}
      </ol>
    </>
  );
}
