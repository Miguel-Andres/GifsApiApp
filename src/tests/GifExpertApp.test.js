import { shallow } from "enzyme"
import GifExpertApp from "../GifExpertApp"

describe('test <GifExpertApp/>', () => { 

    test('debe mostrarse el componente', () => { 
        
        const wrapper = shallow(<GifExpertApp/>)

        expect(wrapper).toMatchSnapshot()


     })

     test('debe de mostrar una lista de category', () => { 

        const categories = ["jordan","cr7"]
        const wrapper = shallow(<GifExpertApp defaultCategories={categories}/>)

        expect(wrapper).toMatchSnapshot()
        expect(wrapper.find("GifGrid").length).toBe(categories.length)
      })
 })