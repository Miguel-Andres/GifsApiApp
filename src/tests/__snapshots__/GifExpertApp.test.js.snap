// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test <GifExpertApp/> debe de mostrar una lista de category 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="jordan"
      key="jordan"
    />
    <GifGrid
      category="cr7"
      key="cr7"
    />
  </ol>
</Fragment>
`;

exports[`test <GifExpertApp/> debe mostrarse el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
